### PR TITLE
hwmv2: port all sparc SoCs and boards

### PR DESCRIPTION
**Pull request by nordicjm**
boards: arm: qemu_leon3: Convert to v2
    
    Converts the board to hwmv2
    

    boards: arm: gr716a: Convert to v2
    
    Converts the board to hwmv2
    

    soc: gr716a: Port to HWMv2
    
    Ports the gr716a SoC configuration to hardware model version 2
    

    boards: arm: generic_leon3: Convert to v2
    
    Converts the board to hwmv2
    

    soc: leon3: Port to HWMv2
    
    Ports the leon3 SoC configuration to hardware model version 2

Doc/compliance likely to fail due to maintainers file not being updated

### CODE REVIEW

**Review by gmarull — COMMENTED** [review]
add board doc entries

File: soc/v2/leon3/Kconfig [hunk 1]
```diff
@@ -2,6 +2,9 @@
 # SPDX-License-Identifier: Apache-2.0
 
 config SOC_LEON3
-	bool "A LEON3 SOC which you can configure"
+	bool
```
Comment (tejlmand) [inline soc/v2/leon3/Kconfig:5]: I believe we still require a `Kconfig.soc` so the soc tree can be reused in sysbuild (and potential other places in addition).

File: soc/v2/gr716a/Kconfig [hunk 2]
```diff
@@ -2,7 +2,10 @@
 # SPDX-License-Identifier: Apache-2.0
 
 config SOC_GR716A
-	bool "GR716A LEON3 fault-tolerant microcontroller"
+	bool
```
Comment (tejlmand) [inline soc/v2/gr716a/Kconfig:5]: we still need a Kconfig.soc which defines
```
config SOC_GR716A
       bool
```

File: boards/v2/gaisler/leon/gr716a_mini/Kconfig.gr716a_mini [hunk 3]
```diff
@@ -3,5 +3,6 @@
 # SPDX-License-Identifier: Apache-2.0
 
 config BOARD_GR716A_MINI
-	bool "GR716-MINI Development Board"
-	depends on SOC_GR716A
+	bool
+	default y
```
Comment (tejlmand) [inline boards/v2/gaisler/leon/gr716a_mini/Kconfig.gr716a_mini:7]:
```suggestion
```

File: boards/v2/qemu/qemu_leon3/board.yml [hunk 4]
```diff
@@ -0,0 +1,5 @@
+board:
+  name: qemu_leon3
```
Comment (tejlmand) [inline boards/v2/qemu/qemu_leon3/board.yml:2]: I think the `qemu` should become a single board where one select the SoC as `qemu/leon3`.

Ref: https://discord.com/channels/720317445772017664/1192088982616539210/1192826204122787890 and following comments.

**Review by tejlmand — CHANGES_REQUESTED** [review]

File: boards/v2/qemu/qemu_leon3/board.yml (hunk 4 above)
Comment (nordicjm) [reply boards/v2/qemu/qemu_leon3/board.yml:2]: That's going to be one huge mess of a directory for files and documentation

File: soc/v2/leon3/Kconfig (hunk 1 above)
Comment (nordicjm) [reply soc/v2/leon3/Kconfig]: Fixed

File: soc/v2/gr716a/Kconfig (hunk 2 above)
Comment (nordicjm) [reply soc/v2/gr716a/Kconfig]: Fixed

File: soc/v2/gr716a/Kconfig.soc [hunk 5]
```diff
@@ -0,0 +1,8 @@
+# Copyright (c) 2019-2020 Cobham Gaisler AB
+# SPDX-License-Identifier: Apache-2.0
+
+config SOC_GR716A
+	bool
+
+config SOC
+        default "gr716a" if SOC_GR716A
```
Comment (tejlmand) [inline soc/v2/gr716a/Kconfig.soc:8]: should be a tab here:
```suggestion
	default "gr716a" if SOC_GR716A
```

File: boards/v2/qemu/qemu_leon3/Kconfig.qemu_leon3 [hunk 6]
```diff
@@ -3,7 +3,7 @@
 # SPDX-License-Identifier: Apache-2.0
 
 config BOARD_QEMU_LEON3
-	bool "QEMU LEON3 target"
-	depends on SOC_LEON3
+	bool
+	select SOC_LEON3
 	select QEMU_TARGET
 	select CPU_HAS_FPU
```
Comment (tejlmand) [inline boards/v2/qemu/qemu_leon3/Kconfig.qemu_leon3:9]: those are Zephyr specifics and not re-usable when sourcing Kconfig tree outside, for example in sysbuild.

Should go to the regular `[boards/v2/qemu/qemu_leon3/Kconfig` file.

File: boards/v2/qemu/qemu_leon3/board.yml (hunk 4 above)
Comment (tejlmand) [reply boards/v2/qemu/qemu_leon3/board.yml:2]: See continued discussion and agreement on Discord.

Just want to point out that board with full identifier now becomes: `qemu_leon3/leon3`.

File: boards/v2/gaisler/generic_leon3/Kconfig.generic_leon3 [hunk 7]
```diff
@@ -3,5 +3,5 @@
 # SPDX-License-Identifier: Apache-2.0
 
 config BOARD_GENERIC_LEON3
-	bool "Generic LEON3 system"
-	depends on SOC_LEON3
+	bool
```
Comment (tejlmand) [inline boards/v2/gaisler/generic_leon3/Kconfig.generic_leon3:6]: `bool` can be omitted.

```suggestion
```

File: boards/v2/qemu/qemu_leon3/Kconfig.qemu_leon3 [hunk 8]
```diff
@@ -3,7 +3,7 @@
 # SPDX-License-Identifier: Apache-2.0
 
 config BOARD_QEMU_LEON3
-	bool "QEMU LEON3 target"
-	depends on SOC_LEON3
+	bool
```
Comment (tejlmand) [inline boards/v2/qemu/qemu_leon3/Kconfig.qemu_leon3:6]: `bool` can be omitted.

```suggestion
```

**Review by tejlmand — CHANGES_REQUESTED** [review]
mostly minor nits left. 
Nice progress :+1:

**Comment by nordicjm** [comment]
> Just want to point out that board with full identifier now becomes: qemu_leon3/leon3.

Do you mean `qemu_leon3/qemu_leon3`? `qemu_leon3` is the board and `qemu_leon3` is the SoC, as `leon3` is a valid SoC for the real hardware, `qemu_leon3` is the emulated hardware which is not real

**Comment by tejlmand** [comment]
> > Just want to point out that board with full identifier now becomes: `qemu_leon3/leon3`.
> 
> Do you mean `qemu_leon3/qemu_leon3`? `qemu_leon3` is the board and `qemu_leon3` is the SoC, as `leon3` is a valid SoC for the real hardware, `qemu_leon3` is the emulated hardware which is not real

according to this:
https://github.com/zephyrproject-rtos/zephyr/blob/772483b7709e0f7662726c40a5ae44fe8aa00301/boards/v2/qemu/qemu_leon3/board.yml#L1-L5

the full identifier will be `qemu_leon3/leon3`, so if that should have been `qemu_leon3/qemu_leon3`, then something is wrong.

File: boards/v2/qemu/qemu_leon3/Kconfig.qemu_leon3 (hunk 8 above)
Comment (nordicjm) [reply boards/v2/qemu/qemu_leon3/Kconfig.qemu_leon3]: Fixed

File: boards/v2/gaisler/generic_leon3/Kconfig.generic_leon3 (hunk 7 above)
Comment (nordicjm) [reply boards/v2/gaisler/generic_leon3/Kconfig.generic_leon3]: Fixed

File: boards/v2/qemu/qemu_leon3/Kconfig.qemu_leon3 (hunk 6 above)
Comment (nordicjm) [reply boards/v2/qemu/qemu_leon3/Kconfig.qemu_leon3]: Fixed

File: soc/v2/gr716a/Kconfig.soc (hunk 5 above)
Comment (nordicjm) [reply soc/v2/gr716a/Kconfig.soc]: Fixed

File: boards/v2/qemu/qemu_leon3/Kconfig.qemu_leon3 [hunk 9]
```diff
@@ -0,0 +1,8 @@
+# Copyright (c) 2019-2020 Cobham Gaisler AB
+#
+# SPDX-License-Identifier: Apache-2.0
+
+config BOARD_QEMU_LEON3
+	select SOC_LEON3
+	select QEMU_TARGET
+	select CPU_HAS_FPU
```
Comment (tejlmand) [inline boards/v2/qemu/qemu_leon3/Kconfig.qemu_leon3:8]: those are still here, ref: https://github.com/zephyrproject-rtos/zephyr/pull/67475#discussion_r1453600423

File: boards/v2/qemu/qemu_leon3/Kconfig.qemu_leon3 (hunk 6 above)
Comment (tejlmand) [reply boards/v2/qemu/qemu_leon3/Kconfig.qemu_leon3]: Not fixed.

File: boards/v2/qemu/qemu_leon3/Kconfig.qemu_leon3 (hunk 9 above)
Comment (nordicjm) [reply boards/v2/qemu/qemu_leon3/Kconfig.qemu_leon3]: Removed this time

**Review by tejlmand — APPROVED** [review]

**Pull request merged**